components/brave_new_tab_ui/components/search/SearchEngineIcon.tsx

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from "react";
+import { useUnpaddedImageUrl } from "../../../brave_news/browser/resources/shared/useUnpaddedImageUrl";
+import { icon } from "@brave/leo/tokens/css";
+import styled from "styled-components";
+
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+const hide = { opacity: 0 }
+function SearchEngineIcon(props: React.DetailedHTMLProps<React.ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement>) {
+  const { src: oldSrc, ...rest } = props
+  const [loaded, setLoaded] = React.useState(false)
+  const onLoad = React.useCallback(() => setLoaded(true), []);
+  const src = useUnpaddedImageUrl(props.src, onLoad, true)
+
+  React.useEffect(() => {
+    setLoaded(false)
+  }, [oldSrc])
+  return <img {...rest} style={loaded ? undefined : hide} src={src} />
```

Comment:
Mmmh, so at the moment, we only use Suspense when lazily loading a component I think. You are meant to be able to use it with fetches, but I've never really looked into it before.

Reply:
edit: just had a look - seems like if you `throw` a promise React will resume rendering when the promise resolves :fearful: 

I think for this, we probably don't want to use a `Suspense` here, as I want to render the image (just transparently) so that the space is reserved and the page doesn't jump around, but I can definitely think of some uses for it!

Reply:
I see. Thanks a bunch for investigation !